File: app/src/main/java/com/icure/kses/modoo/options/WishlistActivity.java
package com.icure.kses.modoo.options;

import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.facebook.drawee.view.SimpleDraweeView;
import com.icure.kses.modoo.R;
import com.icure.kses.modoo.fragments.ImageListFragment;
import com.icure.kses.modoo.product.ItemDetailsActivity;
import com.icure.kses.modoo.utility.ModooDataUtils;
import com.icure.kses.modoo.vo.ModooItemList;

import java.util.ArrayList;

import static com.icure.kses.modoo.fragments.ImageListFragment.STRING_ITEM_CODE;

public class WishlistActivity extends AppCompatActivity {
    private static Context mContext;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.layout_recylerview_list);
        mContext = WishlistActivity.this;

        ModooDataUtils modooDataUtils = new ModooDataUtils();
        ArrayList<ModooItemList> wishlist = modooDataUtils.getWishlist();
        RecyclerView recyclerView = (RecyclerView)findViewById(R.id.recyclerview);
        RecyclerView.LayoutManager recylerViewLayoutManager = new LinearLayoutManager(mContext);

        recyclerView.setLayoutManager(recylerViewLayoutManager);
        recyclerView.setAdapter(new SimpleStringRecyclerViewAdapter(recyclerView, wishlist));
    }

    public static class SimpleStringRecyclerViewAdapter
            extends RecyclerView.Adapter<WishlistActivity.SimpleStringRecyclerViewAdapter.ViewHolder> {

        private ArrayList<ModooItemList> mWishlist;
        private RecyclerView mRecyclerView;

        public static class ViewHolder extends RecyclerView.ViewHolder {
            public final View mView;
            public final SimpleDraweeView mImageView;
            public final LinearLayout mLayoutItem;
            public final ImageView mImageViewWishlist;
            public ViewHolder(View view) {
                super(view);
                mView = view;
                mImageView = (SimpleDraweeView) view.findViewById(R.id.image_wishlist);
                mLayoutItem = (LinearLayout) view.findViewById(R.id.layout_item_desc);
                mImageViewWishlist = (ImageView) view.findViewById(R.id.ic_wishlist);
            }
        }

        public SimpleStringRecyclerViewAdapter(RecyclerView recyclerView, ArrayList<ModooItemList> wishlist) {
            mWishlist = wishlist;
            mRecyclerView = recyclerView;
        }

        @Override
        public WishlistActivity.SimpleStringRecyclerViewAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_wishlist_item, parent, false);
            return new WishlistActivity.SimpleStringRecyclerViewAdapter.ViewHolder(view);
        }

        @Override
        public void onViewRecycled(ViewHolder holder) {
            if (holder.mImageView.getController() != null) {
                holder.mImageView.getController().onDetach();
            }
            if (holder.mImageView.getTopLevelDrawable() != null) {
                holder.mImageView.getTopLevelDrawable().setCallback(null);
//                ((BitmapDrawable) holder.mImageView.getTopLevelDrawable()).getBitmap().recycle();
            }
        }

        @Override
        public void onBindViewHolder(final WishlistActivity.SimpleStringRecyclerViewAdapter.ViewHolder holder, final int position) {
            final Uri uri = Uri.parse(mWishlist.get(position).thumbUrl);
            holder.mImageView.setImageURI(uri);
            holder.mLayoutItem.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent intent = new Intent(mContext, ItemDetailsActivity.class);
                    intent.putExtra(ImageListFragment.STRING_ITEM_CODE, mWishlist.get(position).itemCode);
                    intent.putExtra(ImageListFragment.STRING_IMAGE_URI, mWishlist.get(position).repImageUrl);
                    mContext.startActivity(intent);
                }
            });

            //Set click action for wishlist
            holder.mImageViewWishlist.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    ModooDataUtils modooDataUtils = new ModooDataUtils();
                    modooDataUtils.removeWishlist(position);
                    notifyDataSetChanged();
                }
            });
        }

        @Override
        public int getItemCount() {
            if(mWishlist == null)
                return 0;
            return mWishlist.size();
        }
    }
}

File: app/src/main/java/com/icure/kses/modoo/options/WishlistActivity.kt
package com.icure.kses.modoo.options

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.icure.kses.modoo.R
import com.icure.kses.modoo.activity.ModooItemDetailsActivity
import com.icure.kses.modoo.fragments.HomeListFragment
import com.icure.kses.modoo.utility.ModooDataUtils
import com.icure.kses.modoo.vo.ModooListItem
import kotlinx.android.synthetic.main.layout_recylerview_list.*
import java.util.*

class WishlistActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.layout_recylerview_list)
        mContext = this

        val recylerViewLayoutManager: RecyclerView.LayoutManager = LinearLayoutManager(this)
        recyclerview.layoutManager = recylerViewLayoutManager
        recyclerview.adapter = SimpleStringRecyclerViewAdapter(ModooDataUtils.wishlist)
    }

    class SimpleStringRecyclerViewAdapter(private val mWishlist: ArrayList<ModooListItem>) : RecyclerView.Adapter<SimpleStringRecyclerViewAdapter.ViewHolder>() {

        class ViewHolder(val mView: View) : RecyclerView.ViewHolder(mView) {
            val mImageView: ImageView
            val mLayoutItem: LinearLayout
            val mImageViewWishlist: ImageView

            init {
                mImageView = mView.findViewById<View>(R.id.image_wishlist) as ImageView
                mLayoutItem = mView.findViewById<View>(R.id.layout_item_desc) as LinearLayout
                mImageViewWishlist = mView.findViewById<View>(R.id.ic_wishlist) as ImageView
            }
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.layout_wishlist_item, parent, false)
            return ViewHolder(view)
        }

        override fun onViewRecycled(holder: ViewHolder) {}

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            val uri = Uri.parse(mWishlist[position].thumbUrl)
            Glide.with(mContext!!)
                    .load(uri)
                    .into(holder.mImageView)
            holder.mLayoutItem.setOnClickListener {
                val intent = Intent(mContext, ModooItemDetailsActivity::class.java)
                intent.putExtra(HomeListFragment.STRING_ITEM_CODE, mWishlist[position].itemCode)
                intent.putExtra(HomeListFragment.STRING_IMAGE_URI, mWishlist[position].repImageUrl)
                mContext?.startActivity(intent)
            }

            //Set click action for wishlist
            holder.mImageViewWishlist.setOnClickListener {
                val modooDataUtils = ModooDataUtils()
                modooDataUtils.removeWishlist(position)
                notifyDataSetChanged()
            }
        }

        override fun getItemCount(): Int {
            return mWishlist?.size
        }
    }

    companion object{
        var mContext:Context? = null
    }
}